https.go

```diff
@@ -69,11 +69,11 @@ func (proxy *ProxyHttpServer) connectDial(network, addr string) (c net.Conn, err
 }
 
 func (proxy *ProxyHttpServer) dialContext(ctx *ProxyCtx, network, addr string) (c net.Conn, err error) {
-	if proxy.Tr.DialContext != nil {
-		pctx := context.WithValue(context.Background(), ProxyContextKey, ctx)
-		return proxy.Tr.DialContext(pctx, network, addr)
+	if proxy.Tr.DialContext == nil {
```

Comment:
Ditto. Shouldn't this function and the other changed one in this file behave the same after this logic inversion?

Reply:
Correct! This is mostly just a consistency thing. I thought it was easier to read the dialer chain logic if all of the `if` case statements read then same, and just landed on the `==` logic.